state/machine.go

```diff
@@ -217,7 +217,10 @@ type instanceData struct {
 // upgradeSeriesLock holds the attributes relevant to lock a machine during a
 // series update of a machine
 type upgradeSeriesLock struct {
-	Id string
+	Id         string   `bson:"machineid"`
+	ToSeries   string   `bson:"toseries"`
+	FromSeries string   `bson:"fromseries"`
+	Units      []string `bson:"units"`
```

Comment:
Would it be good to have a `status`  field added which is populated with some string enum (of legit statuses) value? The Uniter is interested in the status. Maybe something like below.

```go
//go:generate stringer -type SeriesUpgradeStatus
const (
	Preparing SeriesUpgradeStatus = iota
	Prepared
	Complete
)
```
``` go
 type upgradeSeriesLock struct {
 ...
 status  UpgradeSeriesStatus `bson: "status"`
```
@hmlanigan we should link up on this so that we can determine how the units behave with respect to this.